zaza/model.py

```diff
@@ -1420,8 +1420,20 @@ def f(x):
     async def _unit_status():
         app = unit_name.split("/")[0]
         model_status = await async_get_status()
-        v = model_status.applications[app]['units'][unit_name][
-            'workload-status']['status']
+        if model_status.applications[app]['units']:
+            v = model_status.applications[app]['units'][unit_name][
+                'workload-status']['status']
+        else:
+            # For when the unit is a subordinate we need to get it's
+            # leader, and then get the status for the subordinate from it's
+            # unit
+            lead_app_name = model_status.applications[app]['subordinate-to'][0]
```

Comment:
Given the possibility of multiple "lead apps", it is probably worth considering adding a "subordinate principal" argument to this function, and iterating over the applications. I'd suggest that, when this argument is none, that we grab the first principal as you've done here, but if the principal name is provided, we use that one.

Reply:
updated as suggested